src/Microsoft.Android.Sdk.ILLink/PreserveRegistrations.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Linq;
+
+using Java.Interop.Tools.Cecil;
+
+using Mono.Cecil;
+using Mono.Linker;
+using Mono.Linker.Steps;
+using MonoDroid.Tuner;
+
+namespace Microsoft.Android.Sdk.ILLink
+{
+	class PreserveRegistrations : BaseSubStep
+	{
+		delegate void AddPreservedMethodDelegate (AnnotationStore store, MethodDefinition key, MethodDefinition method);
+
+		static readonly AddPreservedMethodDelegate addPreservedMethod;
+
+		readonly TypeDefinitionCache cache;
+
+		public PreserveRegistrations (TypeDefinitionCache cache) => this.cache = cache;
+
+		static PreserveRegistrations ()
+		{
+			// temporarily use reflection to get void AnnotationStore::AddPreservedMethod (MethodDefinition key, MethodDefinition method)
+			// this can be removed once we have newer Microsoft.NET.ILLink containing https://github.com/mono/linker/commit/e6dadc995a834603e1178f9a1918f0ae38056b29
+			var method = typeof (AnnotationStore).GetMethod ("AddPreservedMethod", new Type [] { typeof (MethodDefinition), typeof (MethodDefinition) });
+			addPreservedMethod = (AddPreservedMethodDelegate)(method != null ? Delegate.CreateDelegate (typeof (AddPreservedMethodDelegate), null, method, false) : null);
```

Comment:
Should we also update the version we're using?

https://github.com/xamarin/xamarin-android/blob/7eb6d69f61136f3bf7218e4e754e7335df9a093e/src/Microsoft.Android.Sdk.ILLink/Microsoft.Android.Sdk.ILLink.csproj#L10

![image](https://user-images.githubusercontent.com/840039/95085681-2dc38d80-06e5-11eb-88b8-0d9cecc2f1ac.png)

https://github.com/mono/linker/commit/e6dadc995a834603e1178f9a1918f0ae38056b29 is ~3 days ago, we might need to ask to get it backported to this branch: https://github.com/mono/linker/tree/release/5.0

Reply:
The fix is not going to 5.0 branch, so we don't need to update till we switch to 6.0 in future. At least not for this fix.